lib/views/pay/pay_view.dart

```diff
@@ -1,102 +1,214 @@
 import 'package:clnapp/api/api.dart';
 import 'package:clnapp/components/buttons.dart';
+import 'package:clnapp/model/app_model/decode_invoice.dart';
 import 'package:clnapp/model/app_model/pay_invoice.dart';
 import 'package:clnapp/utils/app_provider.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
+import 'numberpad_view.dart';
 
 class PayView extends StatefulWidget {
   final AppProvider provider;
 
   const PayView({Key? key, required this.provider}) : super(key: key);
+
   @override
   State<PayView> createState() => _PayViewState();
 }
 
 class _PayViewState extends State<PayView> {
-  String state = "";
   String? boltString;
-  int? amountMsat;
-  AppPayInvoice? paymentResponse;
-  String? error;
+  final _invoiceController = TextEditingController();
+  late String display;
```

Comment:
```suggestion
  late String display;
```

Mh maybe null, but why you add the late here?

Reply:
In this view particularly handles the invoices paid. So this value will be initialised late but can't be null.